client/events.go

```diff
@@ -46,3 +47,48 @@ func (cli *Client) Events(ctx context.Context, options types.EventsOptions) (io.
 	}
 	return serverResponse.body, nil
 }
+
+// EventsWithHandler decodes the stream of events handing them off to the appropriate
+// handlers for processing. It's up to the caller to stop the processing by canceling
+// the context. The error channel will be closed once the end of the input stream has been
+// reached. If an error is received all processing will be stopped.
+func (cli *Client) EventsWithHandler(ctx context.Context, options types.EventsOptions, handler eventtypes.EventHandler) <-chan error {
+	errChan := make(chan error)
```

Comment:
The entire goroutine will block forever and leak if the caller does not listen on `errChan`.